### PR TITLE
Add basic support for Read Attributes Response in chip-tool

### LINKED ISSUE #1055: Need to support sending commands with data
#### Problem
Right now the only API we seem to have for encoding a command is `chipZclEncodeZclHeader`.  This is ok for commands that just have a cluster and command id, but not enough for commands (like "read attributes" or "configure reporting") that need to have some extra information (like which attributes we want to read or report on.

 #### Proposed Solution
Add an encoding of the relevant information in the right format for the code in `gen-global-command-handler.c` to work with it.

@tecimovic @woody-apple @bhaskar-apple @rwalker-apple

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
#### Problem
Can't issue Read Attributes commands via chip-tool and get responses.

 #### Summary of Changes
Add some encoder APIs for encoding Read Attributes commands and implement some very basic processing of the Read Attributes Response command in chip-tool

fixes https://github.com/project-chip/connectedhomeip/issues/1055 sort of...

### CODE REVIEW

**Comment by bzbarsky-apple** [comment]
Note: this is on top of https://github.com/project-chip/connectedhomeip/pull/1915 for now, hence draft.

**Review by rwalker-apple — APPROVED** [review]

**Review by kedars — APPROVED** [review]

**Review by bhaskar-apple — APPROVED** [review]

File: examples/chip-tool/main.cpp [hunk 1]
```diff
@@ -175,7 +210,8 @@ void ShowUsage(const char * executable)
             "    echo device-ip-address device-port\n"
             "    off device-ip-address device-port endpoint-id\n"
             "    on device-ip-address device-port endpoint-id\n"
-            "    toggle device-ip-address device-port endpoint-id\n",
+            "    toggle device-ip-address device-port endpoint-id\n"
+            "    read-onoff device-ip-address device-port endpoint-id\n",
```
Comment (woody-apple) [inline examples/chip-tool/main.cpp:214]: Maybe we can make read a top level command, and have on-off be a subcommand? (Fine with followup issue)

**Review by woody-apple — APPROVED** [review]

File: examples/chip-tool/main.cpp (hunk 1 above)
Comment (bzbarsky-apple) [reply examples/chip-tool/main.cpp:214]: Not a bad idea.  Filed https://github.com/project-chip/connectedhomeip/issues/1985

File: examples/chip-tool/main.cpp [hunk 2]
```diff
@@ -115,14 +115,49 @@ static void HandleDataModelMessage(System::PacketBuffer * buffer)
...
     // We expect no other bits to be set.
     VerifyOrExit(message[0] == 8, printf("Unexpected frame control byte: 0x%02x\n", message[0]));
     VerifyOrExit(message[1] == 1, printf("Unexpected sequence number: %d\n", message[1]));
-    VerifyOrExit(message[2] == 0x0b, printf("Unexpected command 0x%02x; expected Default Response", message[2]));
-    printf("Got default response to command '0x%02x' for cluster '0x%02x'.  Status is '0x%02x'.\n", message[3], frame.clusterId,
-           message[4]);
+
+    switch (message[2])
+    {
+    case 0x0b: {
```
Comment (BroderickCarlin) [inline examples/chip-tool/main.cpp:126]: Magic number?

Comment (bzbarsky-apple) [reply examples/chip-tool/main.cpp:126]: It's the command id for the Default Response command.  Will document better.

**Comment by bzbarsky-apple** [comment]
@BroderickCarlin @andy31415 @jelderton How does this look to you?

**Review by andy31415 — APPROVED** [review]

**Comment by woody-apple** [comment]
@jelderton @saurabhst @BroderickCarlin ?

**Comment by bzbarsky-apple** [comment]
@BroderickCarlin @jelderton could you take a look please?

**Review by BroderickCarlin — APPROVED** [review]

**Pull request merged**